### ISSUE TITLE
Add project to packages.fhir.org

### ISSUE THREAD

**Issue by peturv** [commonprofiles-care/fhir#54]
We need to add a PR to the ig-registry project to be included in the central registry.

**Comment by peturv** [comment]
Until this is done we need to refer to the dependency by uri:
```
dependencies:
  care.commonprofiles.fhir:
    uri: https://commonprofiles.care/fhir
    version: 1.0.0
```

**Comment by peturv** [comment]
I am currently waiting for my request for the hl7 se profiles, on which we depend on, being accepted into the package repository. Once that is done, I'll put in a request for commonprofiles.

**Comment by vjernelov** [comment]
We have a dependency here to HL7 Sweden's publication of their package. Pétur is pushing for this to move along smoothly.

**Comment by vjernelov** [comment]
Vadim checks with his old colleagues at Firely to understand why things don't show up as expected where they're supposed to.

**Comment by niclas86** [comment]
@vadi2 when do you think you have time to look at this? @claudiaehr is in need of this :)

**Comment by vadi2** [comment]
Hey! Apologies for the delay. We need to follow the DYI part in https://registry.fhir.org/submit to make this work: and it's essentially creating, hosting, and maintaining an RSS feed.

**Comment by vadi2** [comment]
Here is an RSS feed to get us started with:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:fhir="http://hl7.org/fhir/feed" version="2.0">
  <channel>
    <title>HL7 Sweden Packages</title>
    <description>New Packages published by HL7 Sweden</description>
    <link>http://hl7.se/packages/package-feed.xml</link>
    <generator>HL7 Sweden Publication tooling</generator>
    <lastBuildDate>Tue, 30 May 2023 08:31:09 +0000</lastBuildDate>
    <atom:link href="http://hl7.se/packages/package-feed.xml" rel="self" type="application/rss+xml"/>
    <pubDate>Tue, 30 May 2023 08:31:09 +0000</pubDate>
    <language>en</language>
    <ttl>600</ttl>
    <item>
      <title>hl7se.fhir.base#1.0.0</title>
      <description>
        HL7 Sweden base profiles
      </description>
      <link>http://hl7.se/fhir/ig/base/1.0.0/package.tgz</link>
      <guid isPermaLink="true">http://hl7.se/fhir/ig/base/1.0.0/package.tgz</guid>
      <dc:creator>HL7 Sweden</dc:creator>
      <fhir:version>4.0.1</fhir:version>
      <fhir:kind>IG</fhir:kind>
      <pubDate>Thu, 23 March 2023 00:00:00 +0000</pubDate>
      <fhir:details>
        Publication run at 08/03/2023 by N/A using vVersion 1.2.13
      </fhir:details>
    </item>
  </channel>
</rss>
```

The intended location is at http://hl7.se/packages/package-feed.xml, and we can change this to anything we need.

Once the file has been placed there, a new entry in https://github.com/FHIR/ig-registry/blob/master/package-feeds.json#L168 needs to be added and the packages will be live in the registry.

To keep this file up to date with new publications, we'll need to come up with some sort of tooling (or see if other affiliates have worked something out already).

**Comment by claudiaehr** [comment]
@vadi2 when do you think this can be solved?

**Comment by vadi2** [comment]
I've done my part on it, whomever owns hl7.se server now needs to upload the example file I gave. I'm not sure just yet who that party is.

**Comment by claudiaehr** [comment]
@niclas86 @peturv do we know who owns the hl7.se server? I suppose it's HL7 Sweden? Could we maybe contact them in order to get the file uploaded?

**Comment by peturv** [comment]
Sorry, have been away due to illness. Catching up with my email now.
The package feed was put in here: https://hl7.se/fhir/ig/package-feed.xml It isn't the example file you gave, but should be correct output from the latest production build.

Concerning updates of the hl7.se web server I usually contact Arvid Thunholm, but there has been a change to the board, not sure how that affects this. I'll check if there is any change here.

**Comment by peturv** [comment]
Got approval today for the hl7se profiles: https://simplifier.net/packages/hl7se.fhir.base/1.0.0
They are now available on the fhir package server and can be included as dependencies. So now onward to the common profiles releases.

**Comment by vadi2** [comment]
Looks like the same thing now needs to be done for 1.0.1?

**Comment by peturv** [comment]
For hl7se the 1.0.1 has not been formally published, only 1.0.0. But since they are now listed in the package registry new releases should show up almost "automatically". I.e. no PR:s to hl7 should be needed.
But as said, we need to do the same for the commonprofiles.care IG. Now that it can be built on the fhir build server again.

**Comment by niclas86** [comment]
@peturv will publish our IG now

**Comment by vjernelov** [comment]
What's the progress on this @peturv ? We get errors when including the commonprofiles.care package in our own dependencies:

dependencies:
  hl7se.fhir.base: 1.0.0
  care.commonprofiles.fhir:
    uri: https://commonprofiles.care/fhir
    version: 1.0.1

(The HL7se package works fine though)

**Comment by peturv** [comment]
Pull request to the ig-registry has (finally) been submitted from my part

https://github.com/FHIR/ig-registry/pull/214

**Comment by niclas86** [comment]
Waiting for response from HL7 International

**Comment by peturv** [comment]
Pull request has been accepted. It's not up on packages.fhir.org yet, but I will check later today and again after the weekend. If anyone sees it pop up before me, let us know here :)

**Comment by vjernelov** [comment]
The package is automatically retrieved when added to the dependencies in sushi-config.yaml now, so that's a huge step forward! :)

EDIT: Aww, seems like Friday afternoon is hitting hard. It loaded my local cache, not from the registry (where it can't seem to find it still)

**Comment by peturv** [comment]
@claudiaehr Seems we're finally there. Package is deteckted on the packages2.fhir.org server so it is usable and will migrate to packages.fhir.org shortly (I think they sync once per day)

http://packages2.fhir.org/packages/catalog?op=find&name=commonprofiles&pkgcanonical=&canonical=&fhirversion=

**Comment by vjernelov** [comment]
The package version is 1.0.0 in the FHIR registry, but our latest release is 1.0.1.

Shouldn't we make sure the latest is available as well?

**Comment by peturv** [comment]
Was missing update in an index file on our web server. Should be re-scanned by the registry soon:
`1 Packages available. 3 Packages Downloaded. Status = Next Scan in 24min. Last scan was 35min ago `

**Comment by peturv** [comment]
Updated in the registry now :)

**Comment by vjernelov** [comment]
Verified!

**Comment by claudiaehr** [comment]
@peturv The package is still not searchable in Simplifier (I could add the package containing the base profiles by searching for the package-name "hl7se.fhir.base 1.0.0". However, when trying to do the same for the common profiles by searching for the package-name "care.commonprofiles.fhir" as given in http://packages2.fhir.org/packages/catalog?op=find&name=commonprofiles&pkgcanonical=&canonical=&fhirversion= it didn't work...)
![image](https://github.com/commonprofiles-care/fhir/assets/77503664/1356bae8-1486-48f2-bd2b-2a88680d24ae)

**Comment by peturv** [comment]
Both packages2.fhir.org (HL7) and packages.fhir.org (simplifier) are based on the same feed, although simplifier updates slower. I really did think that a weekend would be enough though.

Is there a way to add packages2.fhir.org to the package registries being used in simplifier? Or do they resrict it to only their own (which IS the official registry, so I guess that would be fine as well). I'm trying to find if there is any documentation about the update frequency on packages.fhir.org/simplifier, will get back about that.

**Comment by peturv** [comment]
Found this from Ward Weistra in the Zulip chat: https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Release.20cycle.20for.20https.3A.2F.2Fgithub.2Ecom.2FFHIR.2Fig-registry/near/277769418

> Chiming in from the [registry.fhir.org](http://registry.fhir.org/) / Simplifier perspective.
>
> @Bart Decuypere While maybe not required, I think in practice any guide/package with the intention of being publicly available should want to be published to the FHIR Package Registry ([registry.fhir.org](http://registry.fhir.org/)) so others in the FHIR ecosystem can actually use it (build a specification on top with Simplifier or IG publisher, install locally or to a FHIR server, etc.).
>
> Getting in to the FHIR Package Registry means getting listed in [one of the package feeds](https://github.com/FHIR/ig-registry/blob/master/package-feeds.json), via Simplifier, HL7 publishing process or otherwise, or adding your own feed. Getting a new feed approved to get in that file might take a day or two, as Grahame mentions, but afterwards you can publish any new package, and versions thereof, in that feed. Registered feeds are read by Simplifier and thus [registry.fhir.org](http://registry.fhir.org/), multiple times per day.
So, first time adding a feed: few days, every new package (version) in there: few hours.
>
> On top of that you might want to get your IG (version) published to the manually curated list of the FHIR Implementation Guide Registry (http://fhir.org/guides/registry/). That involves a PR to https://github.com/FHIR/ig-registry/blob/master/fhir-ig-list.json on every publication, so expected ETA of a few days again. But this is only a list for human consumption, not needed for making your guide/package available to the world and all tooling.

I think *this specific point* may be the TLDR:
> So, first time adding a feed: few days, every new package (version) in there: few hours.

I.e. We got approved Friday afternoon. Today it's monday. If there is a manual step involved besides the PR-approval and merge, we might need to give it a tad more time (hours/days not weeks). If nothing has happened or appeared tomorrow, I'll give them a poke.

**Comment by claudiaehr** [comment]
@peturv, as you were mentioning, I think it has to do with how often Simplifier retrieves updates from the fhir package registry. I have contacted Ward from Firely today to check with him. I think nobody else within our group is using Simplifier besides Karolinska so maybe we can close the issues again and I'll solve the rest via direct contact with Firely.

**Comment by peturv** [comment]
@claudiaehr I found an error from the package crawlers registered in the zulip chat and fixed it yesterday. (For those interested; for some reason an empty item had been added to the end of our release item list, in both package and publication feeds). This caused errors due to missing GUID in the...empty items.

Seems to be fixed now: https://simplifier.net/packages/care.commonprofiles.fhir/1.0.1 Not sure if it was my change or your discussion with Ward, but now the packages are everywhere they should be.
